File: Backend/src/Services/CatalogService/CatalogService.API/Dtos/ForCategory/CategoryListDto.cs
using CatalogService.API.Dtos.ForBrand;
using CatalogService.API.Dtos.ForProduct;

namespace CatalogService.API.Dtos.ForCategory
{
    public class CategoryListDto
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public int? ParentCategoryId { get; set; }
        public List<CategoryListDto>?  SubCategories { get; set; }
        public List<FeatureDto>? Features { get; set; }
        public List<BrandListDto>? Brands { get; set; }
        public List<ProductListDto>? Products { get; set; }
    }

}

File: src/Clients/Web/src/dtos/categories/categoryListDto.ts
import { BrandListDto } from "../brands/brandListDto";
import { FeatureDto } from "../features/featureDto";
import { ProductListDto } from "../products/productListDto";

export interface CategoryListDto {
    id: number;
    name: string;
    parentCategoryId: number;
    features: FeatureDto[];
    subCategories?: CategoryListDto[];
    brands?: BrandListDto[];
    products?:ProductListDto[];
}